Tests/ApolloCodegenTests/AnimalKingdomAPI/AnimalKingdomASTCreationTests.swift

```diff
@@ -0,0 +1,589 @@
+import Foundation
+import XCTest
+import Nimble
+import OrderedCollections
+import AnimalKingdomAPI
+@testable import ApolloCodegenLib
+import ApolloCodegenTestSupport
+
+final class AnimalKingdomASTCreationTests: XCTestCase {
+
+  static let frontend = try! ApolloCodegenFrontend()
+
+  static let schema = try! frontend.loadSchema(from: AnimalKingdomAPI.Resources.Schema)
+
+  static let operations = { try! frontend.mergeDocuments(
+    AnimalKingdomAPI.Resources.GraphQLOperations.map {
+      try! frontend.parseDocument(from: $0)
+    }
+  )}()
+
+  static let compilationResult = try! frontend.compile(schema: schema, document: operations)
+
+  override func setUp() {
+    super.setUp()
+  }
+
+  override func tearDown() {
+    super.tearDown()
+  }
+
+  func test__mergedSelections_AllAnimalsQuery_RootQuery__isCorrect() {
+    // given
+    let operation = Self.compilationResult.operations.first { $0.name == "AllAnimalsQuery" }
+    let rootSelectionSet = ASTSelectionSet(selectionSet: operation!.selectionSet,
```

Comment:
i would recommend a `try XCTUnwrap` above rather than `!` since it'll let the rest of your tests keep going if it fails instead of crashing the test suite

(I'll stop myself from doing this for every single one of these but there's a fair few that could be switched to using this)

Reply:
Oh I've never even thought to do that, because I've always used nimble over XCT whenever possible, but you're right that would be great. I'll start using that! Thanks